xmppserver/src/main/java/org/jivesoftware/openfire/ldap/LdapGroupProvider.java

```diff
@@ -124,7 +133,67 @@ private Group getGroupByDN(LdapName groupDN, Set<String> membersToIgnore) throws
             ctx = manager.getContext(baseDN);
             Attributes attrs = ctx.getAttributes(relativeDN, standardAttributes);
 
-            return processGroup(ctx, attrs, membersToIgnore);
+            boolean paging = false;
+            int countPage=0;
+
+            NamingEnumeration<? extends Attribute> i = attrs.getAll();
+            String groupMemberField = manager.getGroupMemberField();
+            String searchString = groupMemberField.contains(";range=")?groupMemberField.substring(0,groupMemberField.indexOf(";range=")):groupMemberField;
+            while (i.hasMore())
+            {
+                Attribute attribute = i.next();
+                String key = attribute.getID().toLowerCase();
+                if (key.startsWith(searchString))
+                {
+                    paging=true;
+                    break;
+                }
+            }
+
+            if (!paging)
+            {
+                return processGroup(ctx, attrs, membersToIgnore);
+            }
+            else
+            {
+                //retrieve first 0-1499
+                Group tmpGroup = processGroup(ctx, attrs, membersToIgnore);
+                Collection<JID> admins = tmpGroup.getAdmins();
+                String description = tmpGroup.getDescription();
+                String name = tmpGroup.getName();
+                ArrayList<JID> members = new ArrayList<JID>();
+                members.addAll(tmpGroup.getMembers());
+                do
+                {
+                    try
+                    {
+                        ArrayList<String> stdAttr = new ArrayList<String>();
+                        for (int n=0;n<standardAttributes.length;n++)
+                        {
+                            if (!standardAttributes[n].contains(manager.getGroupMemberField()))
+                                stdAttr.add(standardAttributes[n]);
+                        }
+                        countPage++;
+                        stdAttr.add(genRangePart(countPage));
+                        attrs = ctx.getAttributes(relativeDN, stdAttr.toArray(new String[stdAttr.size()]));
+                        //retrieve next 1500
+                        tmpGroup=processGroup(ctx, attrs, membersToIgnore);
+                        if (tmpGroup!=null&&tmpGroup.getMembers().size()>0)
+                        {
+                            members.addAll(tmpGroup.getMembers());
+                        }
+                        else // no next found
+                            break;
+                    }
+                    catch (Exception e)
```

Comment:
Do not hide exceptions, by assuming they're caused by an expected condition. Always log them, even if it's just to debug level. I've spent to many weeks chasing a bug that in the end turned out to be hidden by things like `catch (Exception e} { /* can't happen. Ignore. */ }`

Reply:
I'll write to debug log now...